examples/src/main/scala/tofu/example/doobie/TofuDoobieExample.scala

```diff
@@ -0,0 +1,142 @@
+package tofu.example.doobie
+
+import cats.data.ReaderT
+import cats.effect.{ContextShift, Effect, IO, IOApp, Sync}
+import cats.{Apply, Monad}
+import cats.tagless.syntax.functorK._
+import derevo.derive
+import doobie._
+import doobie.implicits._
+import doobie.util.log.LogHandler
+import tofu.doobie.log.{EmbeddableLogHandler, LogHandlerF}
+import tofu.doobie.transactor.Txr
+import tofu.doobie.LiftConnectionIO
+import tofu.higherKind.derived.representableK
+import tofu.lift.{Lift, UnliftIO}
+import tofu.logging.derivation.{loggable, loggingMidTry}
+import tofu.logging.{Logging, LoggingCompanion, Logs}
+import tofu.syntax.context._
+import tofu.syntax.monadic._
+import tofu.syntax.doobie.log.handler._
+import tofu.syntax.doobie.log.string._
+import tofu.{Delay, Tries, WithLocal, WithRun}
+
+// Simple context
+@derive(loggable)
+final case class Ctx(traceId: String)
+
+// Model
+@derive(loggable)
+final case class Person(id: Long, name: String, deptId: Long)
+
+@derive(loggable)
+final case class Dept(id: Long, name: String)
+
+// Person SQL algebra
+@derive(representableK, loggingMidTry)
+trait PersonSql[F[_]] {
+  def init: F[Unit]
+  def create(person: Person): F[Unit]
+  def read(id: Long): F[Option[Person]]
+}
+
+object PersonSql extends LoggingCompanion[PersonSql] {
+  def make[DB[_]: Monad: LiftConnectionIO: EmbeddableLogHandler]: PersonSql[DB] = {
+    EmbeddableLogHandler[DB].embedLift(implicit lh => new Impl)
+  }
+
+  final class Impl(implicit lh: LogHandler) extends PersonSql[ConnectionIO] {
+    def init: ConnectionIO[Unit]                     =
+      lsql"create table if not exists person(id int8, name varchar(50), dept_id int8)".update.run.void
+    def create(p: Person): ConnectionIO[Unit]        =
+      lsql"insert into person values(${p.id}, ${p.name}, ${p.deptId})".update.run.void
+    def read(id: Long): ConnectionIO[Option[Person]] =
+      lsql"select id, name, dept_id from person where id = $id"
+        .query[Person]
+        .option
+  }
+}
+
+// Department SQL algebra
+@derive(representableK, loggingMidTry)
+trait DeptSql[F[_]] {
+  def init: F[Unit]
+  def create(dept: Dept): F[Unit]
+  def read(id: Long): F[Option[Dept]]
+}
+
+object DeptSql extends LoggingCompanion[DeptSql] {
+  def make[DB[_]: Monad: LiftConnectionIO: EmbeddableLogHandler]: DeptSql[DB] = {
+    EmbeddableLogHandler[DB].embedLift(implicit lh => new Impl)
+  }
+
+  final class Impl(implicit lh: LogHandler) extends DeptSql[ConnectionIO] {
+    def init: ConnectionIO[Unit]                   =
+      lsql"create table if not exists department(id int8, name varchar(50))".update.run.void
```

Comment:
I feel like `logsql"..." `would be better here.

Reply:
Let's make a poll in tofu chat.